### PR TITLE
45 displaying questions and subquestions

### PR DESCRIPTION
**Pull request by ShevchenkoKirilo**
Created lesson component, added questions with subquestions loading

### CODE REVIEW

**Comment by ShevchenkoKirilo** [comment]
I moved lesson card styles from lessons-page component to lesson component. They display in browser as previously

File: frontend/src/app/modules/lessons/lesson/lesson.component.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<div class="lesson-card">
+    <div class="lesson-card-content">
+        <div class="photo" (click)="openDialog(lesson.videoId)">
+            <img src="{{ lesson.imgPath }}" alt="thumbnail" />
+            <div class="circle">
+                <img src="../../../../assets/lesson-icons/circle.svg" alt="circle-icon" />
```
[the remark this answers is not in the document]
Comment (tatianahutii) [reply frontend/src/app/modules/lessons/lesson/lesson.component.html:6]: and fix it everywhere

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts [hunk 2]
```diff
@@ -12,6 +8,7 @@ import { LessonsCreateComponent } from '../lessons-create/lessons-create.compone
 export class LessonsPageComponent {
     lessons = [
         {
+            id: 1,
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts:11]: Please this seed data to .utils.ts

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 3]
```diff
@@ -16,6 +16,24 @@ public LessonsService(EasySpeakCoreContext context, IMapper mapper) : base(conte
     {
     }
 
+    public async Task<ICollection<QuestionForLessonDto>> GetQuestionsByLessonIdAsync(int id)
+    {
+        var lesson = await _context.Lessons.Include(l => l.Questions)
+                                           .ThenInclude(q => q.Subquestions)
+                                           .Where(l => l.Id == id)
+                                           .FirstOrDefaultAsync();
```
[the remark this answers is not in the document]
Comment (tatianahutii) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:24]: and please delete Questions from GetAllLessonsAsync method

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/lessons/lesson/lesson.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShevchenkoKirilo) [reply frontend/src/app/modules/lessons/lesson/lesson.component.html]: fixed

File: frontend/src/app/modules/lessons/lesson/lesson.component.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+@Component({
+    selector: 'app-lesson',
+    templateUrl: './lesson.component.html',
+    styleUrls: ['./lesson.component.sass'],
+    template: '<app-loading-spinner></app-loading-spinner>',
+})
+export class LessonComponent {
+    //temporary solution, will be changed to Lesson interface
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    @Input() lesson: any;
```
[the remark this answers is not in the document]
Comment (ShevchenkoKirilo) [reply frontend/src/app/modules/lessons/lesson/lesson.component.ts:20]: Fields in hardcoded list differ a lot from fields we have in database. I think we should resolve this problem before creating an interface for lesson on FE. So this is temporary solution as mentioned in line 18

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 5]
```diff
@@ -16,6 +16,24 @@ public LessonsService(EasySpeakCoreContext context, IMapper mapper) : base(conte
...
+        if (lesson == null)
+        {
+            return new List<QuestionForLessonDto>();
+        }
+
+        var questions = lesson.Questions;
+
+        var questionDtos = _mapper.Map<ICollection<Question>, ICollection<QuestionForLessonDto>>(questions);
+
+        return questionDtos;
```
[the remark this answers is not in the document]
Comment (ShevchenkoKirilo) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs]: fixed

File: frontend/src/app/modules/lessons/lesson/lesson.component.sass [hunk 6]
```diff
@@ -0,0 +1,255 @@
+@import '../../../../styles/colors'
```
[the remark this answers is not in the document]
Comment (ShevchenkoKirilo) [reply frontend/src/app/modules/lessons/lesson/lesson.component.sass]: fixed

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts (hunk 2 above)
Comment (ShevchenkoKirilo) [reply frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts]: fixed

**Review by tatianahutii — DISMISSED** [review]

File: frontend/src/app/modules/lessons/lesson/lesson.component.html [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            <img src="../../../../assets/lesson-icons/expand.svg" alt="expand-icon" />
+        </div>
+    </div>
+    <div class="lesson-card-questions" *ngIf="questions">
+        <div class="question" *ngFor="let question of questions">
+            <p class="question-text">{{ question.topic }}</p>
+            <div class="subquestion" *ngFor="let subquestion of question.subQuestions">
+                <svg class="subquestion-elipse">
+                    <ellipse cx="3" cy="3" rx="3" ry="3" />
+                </svg>
```
[the remark this answers is not in the document]
Comment (ShevchenkoKirilo) [reply frontend/src/app/modules/lessons/lesson/lesson.component.html]: fixed

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 8]
```diff
@@ -16,13 +16,22 @@ public LessonsService(EasySpeakCoreContext context, IMapper mapper) : base(conte
...
     }
 
+    public async Task<ICollection<QuestionForLessonDto>> GetQuestionsByLessonIdAsync(int id)
+    {
+        var questions = _context.Questions.Include(q => q.Subquestions)
+                                          .Where(q => q.LessonId == id);
+
+        var questionDtos = _mapper.Map<ICollection<Question>, ICollection<QuestionForLessonDto>>(await questions.ToListAsync());
+
+        return questionDtos;
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:26]:
```
    var questions = await _context.Questions.Include(q => q.Subquestions)
                                          .Where(q => q.LessonId == id).ToListAsync();

        return _mapper.Map<ICollection<Question>, ICollection<QuestionForLessonDto>>(questions);
```

File: frontend/src/app/modules/lessons/lesson/lesson.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tatianahutii) [reply frontend/src/app/modules/lessons/lesson/lesson.component.ts:20]: Sorry, but I cannot merge this. Create a temporary interface for this

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 9]
```diff
@@ -16,13 +16,22 @@ public LessonsService(EasySpeakCoreContext context, IMapper mapper) : base(conte
     {
     }
 
+    public async Task<ICollection<QuestionForLessonDto>> GetQuestionsByLessonIdAsync(int id)
+    {
+        var questions = _context.Questions.Include(q => q.Subquestions)
```
Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:21]: it would be more readable to use await here

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by Anatolii-Ivanov** [comment]
image

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs (hunk 8 above)
Comment (ShevchenkoKirilo) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs]: fixed

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs (hunk 9 above)
Comment (ShevchenkoKirilo) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs]: fixed

**Review by tatianahutii — APPROVED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**